### PR TITLE
Rename "built-wheels" cache bucket to "source-dists"

### PR DESCRIPTION
**Pull request by zanieb**
This name should lead to less confusion. Unfortunately this is a "breaking cache change" so everyone's cache will be invalidated. I'm not sure if we should support a rename-on-upgrade.

edit: We can make the breaking change next time we bump the version

### CODE REVIEW

File: crates/uv-cache/src/lib.rs [hunk 1]
```diff
@@ -535,6 +535,9 @@ pub enum CacheBucket {
     /// TODO(konstin): The cache policy should be on the source distribution level, the metadata we
     /// can put next to the wheels as in the `Wheels` bucket.
     ///
+    /// The unzipped source distribution is stored in a directory matching the source distribution
+    /// acrhive name.
+    ///
     /// Source distributions are built into zipped wheel files (as PEP 517 specifies) and unzipped
```
Comment (zanieb) [inline crates/uv-cache/src/lib.rs:541]: There is more in this docstring that should change but I do not feel like I have the context on the changes to the cache design to do so.

**Comment by charliermarsh** [comment]
I think this is reasonable but I don't think it merits bumping the cache version. Can we look for an opportunity to do this next time bump the cache?

**Comment by zanieb** [comment]
I dropped the user-facing name change

**Review by charliermarsh — APPROVED** [review]

**Comment by charliermarsh** [comment]
Thanks!

**Pull request merged**